### PR TITLE
utils_misc: Apis to get available nodes and adapt actual machine nodes

### PR DESCRIPTION
**Pull request by hsmj1412**
Signed-off-by: Junxiang Li <junli@redhat.com>

### CODE REVIEW

File: virttest/utils_misc.py [hunk 1]
```diff
@@ -1478,6 +1478,29 @@ def convert_ipv4_to_ipv6(ipv4):
     return converted_ip
 
 
+def cpu_adapter_mod(par_list, std_list):
```
Comment (balamuruhans) [inline virttest/utils_misc.py]: please explain the purpose of this API.

File: virttest/utils_misc.py [hunk 2]
```diff
@@ -1478,6 +1478,29 @@ def convert_ipv4_to_ipv6(ipv4):
...
+                if un in std_list:
+                    tmp_list.append(un)
+            for nd in std_list:
+                if not len(par_list) > len(tmp_list):
+                    break
+                if nd not in tmp_list:
+                    tmp_list.append(nd)
+            if len(par_list) == len(tmp_list):
+                par_list = tmp_list
+    return par_list
```
Comment (balamuruhans) [inline virttest/utils_misc.py]: can `set(par_list).union(set(std_list))` help you in this logic.

File: virttest/utils_misc.py [hunk 3]
```diff
@@ -1718,6 +1739,20 @@ def get_all_node_meminfo(self):
...
+        :return: The ids of node which is available
+        :rtype: builtin.list
+        """
+        node_list = self.online_nodes
+        memory_list = self.get_all_node_meminfo()
+        for nodenum in memory_list:
+            if int(memory_list[nodenum]["MemTotal"]) == 0 and nodenum in node_list:
+                node_list.remove(nodenum)
+        return node_list
+
```
Comment (sathnaga) [inline virttest/utils_misc.py]: @hsmj1412 you can consider this logic to get all nodes with mem, https://github.com/avocado-framework/avocado-vt/pull/1158/commits/b2ed761977efacc5f634c47d791f986f75ac0051 ?
It should be reliable as host provides the info.

Comment (hsmj1412) [reply virttest/utils_misc.py]: It looks better than this PR, thanks a lot!

**Pull request closed without merging**